backend/mem/file_test.go

```diff
@@ -0,0 +1,723 @@
+
+	found, err := file.Exists()
+	s.NoError(err, "exists error not expected")
+	s.True(found)
+
+	s.NoError(file.Delete(), "File was not deleted properly")
+
+	found2, err := file.Exists()
+	s.NoError(err, "exists error not expected")
+	s.False(found2)
+
+}
+
+/*
+TestCopyToLocation copies a file to a location that has
+ been passed in. Succeeds on existence of original file and its
+contents in new location
+*/
+func (s *memFileTest) TestCopyToLocation() {
+
+	newFile, err := s.fileSystem.NewFile("", "/home/foo.txt")
+	s.NoError(err, "Unexpected error creating file")
+
+	expectedText := "hello world!"
+	_, err = s.testFile.Write([]byte(expectedText))
+	s.NoError(err, "Unexpected write error")
+
+	//seeking to the start of the file from the end of the file (offset of -12 takes us to the start)
+	_, err = s.testFile.Seek(-12, 2)
+	s.NoError(err, "Seek unexpectedly threw an error")
+
+	readSlice1 := make([]byte, len(expectedText))
+	readSlice2 := make([]byte, len(expectedText))
+
+	copiedFile, cerr := s.testFile.CopyToLocation(newFile.Location())
+	s.NoError(cerr, "CopyToLocation unexpectedly failed")
+
+	copiedFile.(*File).Touch()
+	s.True(copiedFile != nil)
+	//making sure the path was correctly updated
+	s.EqualValues("/home/test.txt", copiedFile.Path())
+
+	_, err = copiedFile.Read(readSlice1)
+	s.NoError(err, "Unexpected read error.")
+
+	_, err = s.testFile.Read(readSlice2)
+	s.NoError(err, "Unexpected read error.")
+	s.EqualValues(string(readSlice2), string(readSlice1))
+
+}
+
+/*
+TestCopyToLocationOW copies a file to a location that has
+ a file with the same name. Succeeds only on overwrite of file
+ in specified location
+*/
+func (s *memFileTest) TestCopyToLocationOW() {
+
+	newFile, err := s.fileSystem.NewFile("C", "/home/test.txt")
+	s.NoError(err, "Unexpected error creating a file")
+	newFile.(*File).Touch()
+	originalText := "goodbye world!"
+	_, err = newFile.Write([]byte(originalText))
+	s.NoError(err, "Unexpected write error")
+	s.NoError(newFile.Close(), "Unexpected Close error")
+
+	expectedText := "hello world!"
+	_, err = s.testFile.Write([]byte(expectedText))
+	s.NoError(err, "Unexpected write error")
+	s.NoError(s.testFile.Close(), "Unexpected Close error")
+
+	readSlice := make([]byte, len(expectedText))
+	copiedFile, err := s.testFile.CopyToLocation(newFile.Location())
+	s.NoError(err, "CopyToLocation unexpectedly failed")
+	s.True(copiedFile != nil)
+
+	s.EqualValues("/home/test.txt", copiedFile.Path())
+	_, err = copiedFile.Read(readSlice)
+	s.NoError(err, "Unexpected read error.")
+	s.EqualValues("hello world!", string(readSlice))
+
+}
+
+//TestCopyToNilFile ensures that an error is thrown for trying to copy to a nil file
+func (s *memFileTest) TestCopyToNilFile() {
+
+	newFile, err := s.fileSystem.NewFile("", "/nilFile.txt")
+	s.NoError(err, "Unexpected error creating file")
+	newFile = nil
+	err = s.testFile.CopyToFile(newFile)
+	s.Error(err, "Expected error for copying to nil file")
+}
```

Comment:
we can just remove lines 334 and 335.  They aren't doing anything in regards to our test.